Conduct a Chi-square test for independence between two categorical variables. Based on
the p-value and test result, interpret whether there is a significant association.

In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv(r'E:\Code\Maths assignment\tested.csv')

In [3]:
# Survived vs Pclass
import scipy.stats as stats

# Create a contingency table
contingency_table = pd.crosstab(df['Survived'], df['Pclass'])

# Perform Chi-square test
chi2, p, dof, expected = stats.chi2_contingency(contingency_table)

print(f"Chi-square value: {chi2}")
print(f"P-value: {p}")

Chi-square value: 6.693869422819262
P-value: 0.03519206276590605


Since the p-value (0.035) is less than 0.05, we reject the null hypothesis. This means that there is a significant association between Survived and Pclass in the Titanic dataset.